lib/src/data/communication/api/ny_times_repository_impl.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:allthenews/src/data/communication/api/ny_times_repositor.dart';
+import 'package:allthenews/src/di/injector.dart';
+import 'package:allthenews/src/domain/authorization/api_key.dart';
+import 'package:allthenews/src/domain/authorization/api_key_repository.dart';
+import 'package:allthenews/src/domain/model/article.dart';
+import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
+import 'package:pretty_dio_logger/pretty_dio_logger.dart';
+
+class _Urls {
+  static const baseUrl = "https://api.nytimes.com/svc/";
+  static const mostPopular = "mostpopular/v2/emailed/7.json";
+}
+
+class NYTimesRepositoryImpl extends NYTimesRepository {
+  final Dio _dio = inject<Dio>();
+  ApiKeyRepository _apiKeyRepository = inject<ApiKeyRepository>();
+  ApiKey _apiKey;
+
+  NYTimesRepositoryImpl() {
+    _dio.options.baseUrl = _Urls.baseUrl;
+
+    final verboseLogging = kDebugMode;
+
+    _dio.interceptors
+      ..add(
+        InterceptorsWrapper(onRequest: _onRequest),
```

Comment:
await?

Reply:
nie, to tylko przekazanie referencji, nie ma zwracanego typu